MonoZelda/UI/HUDMapWidget.cs

```diff
@@ -54,6 +54,9 @@ public override void Load(ContentManager content)
 
     public override void Update()
     {
+        SetCompassMarkerVisible(PlayerState.HasCompass);
+        SetMapVisible(PlayerState.HasMap);
+
```

Comment:
These should ideally only be called when the value changes, not every single frame, but they're small methods so this is not much of a performance hit.